PowerShell/JumpCloud Module/Private/NestedFunctions/Get-JCResults.ps1

```diff
@@ -220,25 +229,30 @@ Function Get-JCResults {
                 # Add results to results list
                 if ($totalCountHeader) {
                     # Add results to results list
-                    $content = $response.Content
-                    [void]$resultsArray.Add($content)
-                    Write-Debug ("Page: $($i+1) Amount: " + ($content | ConvertFrom-Json).Count)
+                    $content = $response.Content | ConvertFrom-Json
+                    if ($content.Count -le 1) {
+                        [void]$resultsArray.Add($content)
+                    } else {
+                        [void]$resultsArray.AddRange($content)
+                    }
                 } else {
                     # Add results to results list
                     $content = $response.Content | ConvertFrom-Json
                     if ($null -eq $content.results) {
                         [void]$resultsArray.Add($content)
-                        Write-Debug ("Page: $($i+1) Amount: " + ($content | ConvertFrom-Json).Count)
                     } else {
-                        [void]$resultsArray.AddRange($content.results)
-                        Write-Debug ("Page: $($i+1) Amount: " + ($content.results).Count)
+                        if ($($content.results).Count -le 1) {
+                            [void]$resultsArray.Add($content.results)
+                        } else {
+                            [void]$resultsArray.AddRange($content.results)
+                        }
                     }
                 }
             }
         }
     }
     end {
-        if ($totalCountHeader) {
+        if ($parallel -and $totalCountHeader) {
```

Comment:
![Screen Shot 2023-08-01 at 10 44 01 AM](https://github.com/TheJumpCloud/support/assets/54448601/d0f4d1a0-5d96-434c-8d52-94255863c272)
I can confirm this resolves the issue for my older version of PWSH